lib/resultuploaders/s3compatible/s3compatible.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'aws-sdk-s3'
+require './lib/auxiliary/json_helper'
+
+# AutoHCK module
+module AutoHCK
+  # S3compatible class
+  class S3compatible
+    CONFIG_JSON = 'lib/resultuploaders/s3compatible/s3compatible.json'
+
+    include Helper
+
+    attr_reader :url
```

Comment:
Yes, this part is not fully completed yet. Since there is no concept of folders in S3, directly accessing a folder URL may not be feasible. I see two methods that could potentially replace accessing the folder URL:

1. Hosting the bucket as a static website, which requires a lot of upfront configuration. see https://docs.aws.amazon.com/AmazonS3/latest/userguide/HostingWebsiteOnS3Setup.html
2. Creating an index.html file in the folder, which the URL will point to. This index.html will store the URLs of the actual files, and it will be updated every time files are uploaded or deleted.
![image](https://github.com/user-attachments/assets/c86e8434-50fa-4cac-bc98-90407c37005b)


Currently, I plan to choose option 2 (the code will be uploaded later). What do you think?

Reply:
Looks good according to S3 behavior. @akihikodaki What do you think?

Reply:
I suspect you will want to see `report.html` instead of the file list. `report.html` may include the file list or link to one.

Reply:
@akihikodaki ，Thank you for your feedback! I'm a bit unclear about your point. Could you please elaborate a bit more? When you mention report.html, are you referring to result.html（see below)? My intention is indeed to provide a file list, which refers to the URL of the folder containing the test results. Since S3 doesn’t have a concept of folders, I used 'file list' to represent this idea. By accessing that URL, we can retrieve all the results from this test.
![image](https://github.com/user-attachments/assets/e04f0f70-e251-4497-b04d-88ac21bf234d)

Reply:
Yes, I referred to that file. I see the utility of the file list, but I suspect perhaps you may want to see `results.html` first, and navigate to the file list from there.

Reply:
Okay, thank you, I understand your point now. My original intention was indeed to provide the URL of the folder directly. If users need to access result.html, they can obtain it through the URL of this folder. If I change the URL to point to result.html, I will encounter an issue that I haven't figured out how to address appropriately. Specifically, result.html is only generated during the “bin/auto_hck test", but not during "bin/auto_hck install". Additionally, if an exception occurs during the HCK test, result.html might not be generated, and users may want to refer to other log files to identify the issue

Reply:
Fair point about install. Let's keep the index.